src/plugins/data/common/search/aggs/agg_config.ts

```diff
@@ -439,10 +439,14 @@ export class AggConfig {
   }
 
   fieldIsTimeField() {
-    const indexPattern = this.getIndexPattern();
-    if (!indexPattern) return false;
-    const timeFieldName = indexPattern.timeFieldName;
-    return timeFieldName && this.fieldName() === timeFieldName;
+    const defaultTimeField = this.getIndexPattern()?.getTimeField?.()?.name;
+    const defaultTimeFields = defaultTimeField ? [defaultTimeField] : [];
+    const allTimeFields =
+      this.aggConfigs.timeFields && this.aggConfigs.timeFields.length > 0
+        ? this.aggConfigs.timeFields
```

Comment:
will this contain the default one? or we don't want the default one if this was provided ?

Reply:
It overwrites the default one - this is the same logic as `esaggs` uses for building the filter